modules/ghlint-extensions/src/main/kotlin/net/twisterrob/ghlint/model/effectiveScopes.kt

```diff
@@ -0,0 +1,24 @@
+package net.twisterrob.ghlint.model
+
+public val Permissions.effectiveScopes: Set<Scope>
+	get() =
+		mutableSetOf(
+			Scope(Permission.ACTIONS, actions),
+			Scope(Permission.ATTESTATIONS, attestations),
+			Scope(Permission.CHECKS, checks),
+			Scope(Permission.CONTENTS, contents),
+			Scope(Permission.DEPLOYMENTS, deployments),
+			Scope(Permission.ID_TOKEN, idToken),
+			Scope(Permission.ISSUES, issues),
+			Scope(Permission.METADATA, metadata),
+			Scope(Permission.PACKAGES, packages),
+			Scope(Permission.PAGES, pages),
+			Scope(Permission.PULL_REQUESTS, pullRequests),
+			Scope(Permission.REPOSITORY_PROJECTS, repositoryProjects),
+			Scope(Permission.SECURITY_EVENTS, securityEvents),
+			Scope(Permission.STATUSES, statuses)
+		).let { scopes ->
+			scopes + scopes
+				.filter { it.access == Access.WRITE }
+				.map { Scope(it.permission, Access.READ) }
```

Comment:
Did you want to include the defaulted `NONE`s in this list?

```suggestion
			scopes
				.filterNot { it.access == Access.NONE }
			+
			scopes
				.filter { it.access == Access.WRITE }
				.map { Scope(it.permission, Access.READ) }
```

There are some failing unit tests because of this, for me it was unexpected to have nones, but the technically for "effective" permissions they seem correct. Absence of them would probably give as much info.

Reply:
Decided to go for explicitness and resolve them all to `NONE` when relevant.

ee4d75a5e0a8c4428932c591096ce59af78ab29b